File: frontend/apps/admin/src/services/medicine.ts
import request from './request';

// 管理端药品与分类管理接口。
export interface MedicineQuery {
  page?: number;
  size?: number;
  keyword?: string;
  categoryId?: number;
  isPrescription?: number;
  status?: number;
}

export interface Medicine {
  id: number;
  name: string;
  categoryId: number;
  categoryName?: string;
  mainImage: string;
  price: number;
  stock: number;
  sales: number;
  isPrescription: number; // 0: OTC, 1: Rx
  indication: string;
  usageMethod: string;
  contraindication: string;
  expiryDate: string;
  productionDate: string;
  sellerId: number;
  sellerName?: string;
  status: number; // 1: 上架, 0: 下架
  createTime: string;
}

export interface PageResult<T> {
  records: T[];
  total: number;
  size: number;
  current: number;
  pages: number;
}

export interface Result<T> {
  code: number;
  message: string;
  data: T;
}

// 管理端药品分页列表，支持关键字和状态筛选。
export const getAdminMedicineList = (params: MedicineQuery) => {
  return request.get<PageResult<Medicine>>('/medicine/admin/list', { params });
};

// 管理员删除药品。
export const deleteMedicineAdmin = (id: number) => {
  return request.delete(`/medicine/admin/${id}`);
};

// 管理员强制上下架药品。
export const updateMedicineStatusAdmin = (id: number, status: number) => {
  return request.patch(`/medicine/admin/${id}/status?status=${status}`);
};

// 分类接口同时服务于药品筛选和分类管理页面。
export interface Category {
  id: number;
  name: string;
  parentId: number;
  level: number;
  sort: number;
  icon?: string;
}

export const getCategoryList = () => {
  return request.get<Category[]>('/category/list');
};

export const addCategory = (data: Partial<Category>) => {
  return request.post('/category', data);
};

export const updateCategory = (data: Partial<Category>) => {
  return request.put('/category', data);
};

export const deleteCategory = (id: number) => {
  return request.delete(`/category/${id}`);
};

File: backend/zhijian-medicine/src/main/java/com/zhijian/pojo/medicine/entity/Medicine.java
package com.zhijian.pojo.medicine.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 药品实体类。
 */
@Data
@TableName("pms_medicine")
public class Medicine implements Serializable {

    /**
     * 序列化版本号。
     */
    private static final long serialVersionUID = 1L;

    /**
     * 药品 ID。
     */
    @TableId(value = "id")
    private Long id;

    /**
     * 药品名称。
     */
    private String name;

    /**
     * 分类 ID。
     */
    private Long categoryId;

    /**
     * 分类名称。
     */
    @TableField(exist = false)
    private String categoryName;

    /**
     * 主图地址。
     */
    private String mainImage;

    /**
     * 价格。
     */
    private BigDecimal price;

    /**
     * 库存。
     */
    private Integer stock;

    /**
     * 销量。
     */
    private Integer sales;

    /**
     * 规格。
     */
    private String specs;

    /**
     * 是否处方药。
     */
    private Integer isPrescription;

    /**
     * 适应症。
     */
    private String indication;

    /**
     * 用法用量。
     */
    private String usageMethod;

    /**
     * 禁忌。
     */
    private String contraindication;

    /**
     * 有效期。
     */
    private LocalDate expiryDate;

    /**
     * 生产日期。
     */
    private LocalDate productionDate;

    /**
     * 商家 ID。
     */
    private Long sellerId;

    /**
     * 商家名称。
     */
    @TableField(exist = false)
    private String sellerName;

    /**
     * 上下架状态。
     */
    private Integer status;

    /**
     * 创建时间。
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间。
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 逻辑删除标记。
     */
    private Integer deleted;
}
